src/methods/call.js

```diff
@@ -250,7 +251,7 @@ export function OrdinaryCallBindThis(
   thisArgument: Value
 ): NullValue | ObjectValue | AbstractObjectValue | UndefinedValue {
   // 1. Let thisMode be the value of F's [[ThisMode]] internal slot.
-  let thisMode = F.$ThisMode;
+  let thisMode = F instanceof ECMAScriptFunctionValue ? F.$ThisMode : undefined;
```

Comment:
Reading the code below, it seems that the default should be "strict", and then you can remove the branch `|| F instanceof NativeFunctionValue` below, which isn't in the spec anyway.